include/circt/Dialect/OM/OMOpInterfaces.td

```diff
@@ -15,6 +15,102 @@
 
 include "mlir/IR/OpBase.td"
 
+def ClassFieldsLike : OpInterface<"ClassFieldsLike"> {
+  let cppNamespace = "circt::om";
+
+  let description = [{
+    Marks the end of a class-like definition by providing a list of accessible
+    fields.
+  }];
+  let methods = [
+    InterfaceMethod<"Get the class-like field to type map",
+      "std::optional<mlir::Type>", "getFieldType", (ins "mlir::StringAttr":$name)>,
+    InterfaceMethod<"Print class fields like",
+      "void", "printClassFieldsLike", (ins "mlir::OpAsmPrinter &":$printer,
+                                           "llvm::SmallVector<mlir::Value>":$operands),
+      /*methodBody=*/[{}],
+      /*defaultImplementation=*/[{ 
+        // * assumes fieldNames dictionary attr exists
+        // * assumes each name in fieldNames has an entry in fieldTypes
+        //   dictionary attr mapping the name to type
+        // * perFieldAttrs, perFieldLocs optional attr/locs for each field
+        // * each field is emitted on a separate line
+        mlir::Operation *op = $_op.getOperation();
+
+        printer << "(";
+        printer.increaseIndent();
+
+        llvm::SmallVector<mlir::StringAttr> fieldNames = $_op.getFieldNames();
+
+        mlir::ArrayRef<mlir::Attribute> perFieldAttrs;
```

Comment:
Maybe I'm missing something, but do we use attributes on each field? I know the original `om.class.field` has an arbitrary attributes dictionary, but that's because all operations have that, to hold any discardable attributes. I don't think we've actually used per-field attributes for anything, so it might be fine to remove. We do have per-port attributes on HW modules, but I don't know if we need to follow that here unless we have a use case.

Reply:
Yea, I think I was just following the HW code.  Figured it might be useful but none of the test code actually needed it.